fontc_crater/src/main.rs

```diff
@@ -86,10 +86,10 @@ enum SkipReason {
 }
 
 #[allow(clippy::type_complexity)] // come on, it's not _that_ bad
-fn run_all<T: Send, E: Send>(
+fn run_all<T: Send, E: Send, Cx: Sync>(
     targets: Vec<Target>,
-    cache_dir: &Path,
-    runner: impl Fn(&Path, &Target) -> RunResult<T, E> + Send + Sync,
+    context: &Cx,
+    runner: impl Fn(&Cx, &Target) -> RunResult<T, E> + Send + Sync,
 ) -> Result<Vec<(Target, RunResult<T, E>)>, Error> {
```

Comment:
I'm with clippy, this seems over-complicated. Do we really need it to be so generic?

Reply:
it's a legacy of the earlier design, where this one function was responsible for executing various different runners that would produce different types of output. It could be simplified but doesn't feel like a priority...